js/ui/components/oldClubhouse.js

```diff
@@ -0,0 +1,339 @@
+        this._closeButton.connect('clicked', () => { this.emit('close-clubhouse'); })
+
+        Main.layoutManager.addChrome(this._closeButton);
+
+        this._clubhouseWindowActor = null;
+        this._clubhouseNotifyHandler = 0;
+
+        this._visible = true;
+        this._visibleOpenButton = true;
+
+        this._updateVisibility();
+
+        this._updateVisibilityId = getClubhouseWindowTracker().connect('window-changed', this._updateVisibility.bind(this));
+    }
+
+    displayOpenButton(value) {
+        this._visibleOpenButton = value;
+    }
+
+    _updateCloseButtonPosition() {
+        this._closeButton.y = this._openButton.y;
+
+        let clubhouseWindowX = getClubhouseWindowTracker().getWindowX();
+        if (clubhouseWindowX != -1)
+            this._closeButton.x = clubhouseWindowX - this._closeButton.width / 2;
+
+        Clubhouse.clipToMonitor(this._closeButton);
+    }
+
+    _reposition() {
+        let monitor = Main.layoutManager.primaryMonitor;
+        if (!monitor)
+            return;
+
+        let workarea = Main.layoutManager.getWorkAreaForMonitor(monitor.index);
+
+        this._openButton.x = monitor.x + monitor.width - this._openButton.width / 2;
+        this._openButton.y = workarea.y + Math.floor(workarea.height / 2.0 - this._openButton.height / 2.0);
+
+        this._updateCloseButtonPosition();
+    }
+
+    _updateVisibility() {
+        this._closeButton.visible = this._visible && getClubhouseWindowTracker().getWindowX() != -1;
+        this._openButton.visible = this._visible && this._visibleOpenButton &&
+                                   Main.sessionMode.hasOverview && !this._closeButton.visible;
+        this._reposition();
+    }
+
+    setSuggestOpen(suggestOpen) {
+        if (this._openButton.visible)
+            this._openButton.setHighlighted(suggestOpen);
+    }
+
+    setVisible(visible) {
+        this._visible = visible;
+        this._updateVisibility();
+    }
+
+    destroy() {
+        if (this._updateVisibilityId)
+            getClubhouseWindowTracker().disconnect(this._updateVisibilityId);
+        this._closeButton.destroy();
+        this._openButton.destroy();
+    }
+});
+
+var Component = GObject.registerClass({
+}, class OldClubhouseComponent extends Clubhouse.Component {
+    _init() {
+        // Enable hack-mode if we've the old clubhouse and the hack-mode is
+        // disabled because this should be activated to make it work, so hack
+        // mode will be enabled by default for hack 1.0 users
+        const activated = global.settings.get_boolean('hack-mode-enabled');
+        if (!activated && this.getClubhouseApp())
+            global.settings.set_boolean('hack-mode-enabled', true);
+
+        super._init(CLUBHOUSE_IFACE, CLUBHOUSE_ID, CLUBHOUSE_DBUS_OBJ_PATH);
+        this._clubhouseButtonManager = null;
+
+        let keybindingSettings = new Gio.Settings({ schema: SHELL_KEYBINDINGS_SCHEMA });
+        Main.wm.addKeybinding('toggle-clubhouse',
```

Comment:
Could you add the keybinding as well as the `_updateCloseButtonPosition` function and `_visibleOpenButton` variable/attribute as part of a new commit?